js/src/components/shipping-rate-section/estimated-shipping-rates-card/rate-form-modals/add-rate-form-modal.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import RateFormModal from './rate-form-modal.js';
+
+/**
+ * @typedef { import(".~/data/actions").CountryCode } CountryCode
+ * @typedef { import("../typedefs.js").ShippingRateGroup } ShippingRateGroup
+ */
+
+/**
+ * Form to add a new rate for selected country(-ies).
+ *
+ * @param {Object} props
+ * @param {Array<CountryCode>} props.countryOptions Array of country codes, to be used as options in AppCountrySelect.
+ * @param {ShippingRateGroup} props.initialValues Initial values for the form.
+ * @param {(values: ShippingRateGroup) => void} props.onSubmit Called with submitted value.
+ * @param {() => void} props.onRequestClose Callback to close the modal.
+ */
+const AddRateFormModal = ( {
+	countryOptions,
+	initialValues,
+	onSubmit,
+	onRequestClose = noop,
```

Comment:
:scroll: I got myself head-scratching for some time, thinking: "hey [the only use of `AddRateFormModal` we have](https://github.com/woocommerce/google-listings-and-ads/pull/1424/files#diff-5bc59001786c70aecac2fb120aec5a8fa59cfd84a6784ef80ba27a7c14c04c70R155-R164) does not set `onRequestClose`, so why don't we remove it?"

That's already reported in https://github.com/woocommerce/google-listings-and-ads/issues/1439

:nail_care: But I think the default value of `noop` is not needed here, as we don't use it without given `onRequestClose` prop. So maybe it's YAGNI, plus we can fails fast if something is unexpectedly missing.